src/main/java/org/spongepowered/api/item/inventory/InventoryArchetypes.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import com.flowpowered.math.vector.Vector2i;
+import org.spongepowered.api.item.inventory.property.AcceptsItems;
+import org.spongepowered.api.item.inventory.property.InventorySize;
+import org.spongepowered.api.item.inventory.property.SlotIndex;
+import org.spongepowered.api.item.inventory.property.TitleProperty;
+
+/**
+ * An enumeration of {@link InventoryArchetype}s.
+ */
+public class InventoryArchetypes {
+
+    // TODO Dummies
+
+    // Base Archetypes
+
+    // TODO how to start?
+    public static final InventoryArchetype SLOT = null;
+    public static final InventoryArchetype MENU_ROW = null;
+    public static final InventoryArchetype MENU_GRID = null;
+
+    // Vanilla Archetypes
+
+    // Examples
+    public static final InventoryArchetype CHEST = null;
+    {
+        CHEST = InventoryArchetype.builder()
+                .with(MENU_GRID)
+                .property(new InventorySize(9, 3))
+                .property(new TitleProperty("minecraft.tile.chest.name"))
+                .build("minecraft:chest", "Chest");
+    }
+    public static final InventoryArchetype DOUBLE_CHEST = null; // Composite of 2 CHEST
+    {
+        DOUBLE_CHEST = InventoryArchetype.builder()
+                .with(CHEST).with(CHEST)
+                .with(CHEST, 2) // alternative?
+                .property(new InventorySize(9, 6))
+                .property(new TitleProperty("minecraft.tile.chest.name"))
+                .build("minecraft:chest2", "DoubleChest");
+    }
+    public static final InventoryArchetype FURNACE = null;
+    {
+        FURNACE = InventoryArchetype.builder()
+                .with(SLOT.toBuilder().property(new SlotIndex(0)).build("minecraft:furnace_input", "FurnaceInput"))
```

Comment:
As @Zidane said, this is pseudocode. In reality this information is simply slurped into the Archetype impl from the relevant `Lens`.